multisig-ism-metadata-test/tests/harness.rs

```diff
@@ -0,0 +1,164 @@
+use fuels::{prelude::*, tx::ContractId};
+use hyperlane_core::{utils::domain_hash, Checkpoint, H160, H256};
+use sha3::{Digest, Keccak256};
+use test_utils::{bits256_to_h256, h256_to_bits256};
+
+// Load abi from json
+abigen!(
+    TestMultisigIsmMetadata,
+    "out/debug/multisig-ism-metadata-test-abi.json"
+);
+
+const TEST_MAILBOX_ADDRESS: H256 = H256::repeat_byte(0xau8);
+const TEST_MAILBOX_DOMAIN: u32 = 420u32;
+const TEST_CHECKPOINT_ROOT: H256 = H256::repeat_byte(0xbu8);
+const TEST_CHECKPOINT_INDEX: u32 = 69u32;
+const TEST_THRESHOLD: u8 = 4u8;
+const TEST_VALIDATORS: &[H160; 5] = &[
+    H160::repeat_byte(0x1u8),
+    H160::repeat_byte(0x2u8),
+    H160::repeat_byte(0x3u8),
+    H160::repeat_byte(0x4u8),
+    H160::repeat_byte(0x5u8),
+];
+
+async fn get_contract_instance() -> (TestMultisigIsmMetadata, ContractId) {
+    // Launch a local network and deploy the contract
+    let mut wallets = launch_custom_provider_and_get_wallets(
+        WalletsConfig::new(
+            Some(1),             /* Single wallet */
+            Some(1),             /* Single coin (UTXO) */
+            Some(1_000_000_000), /* Amount per coin */
+        ),
+        None,
+        None,
+    )
+    .await;
+    let wallet = wallets.pop().unwrap();
+
+    let id = Contract::deploy(
+        "./out/debug/multisig-ism-metadata-test.bin",
+        &wallet,
+        TxParameters::default(),
+        StorageConfiguration::with_storage_path(Some(
+            "./out/debug/multisig-ism-metadata-test-storage_slots.json".to_string(),
+        )),
+    )
+    .await
+    .unwrap();
+
+    let instance = TestMultisigIsmMetadata::new(id.clone(), wallet);
+
+    (instance, id.into())
+}
+
+fn get_test_checkpoint_and_metadata() -> (Checkpoint, MultisigMetadata) {
+    let checkpoint = Checkpoint {
+        mailbox_address: TEST_MAILBOX_ADDRESS,
+        mailbox_domain: TEST_MAILBOX_DOMAIN,
+        root: TEST_CHECKPOINT_ROOT,
+        index: TEST_CHECKPOINT_INDEX,
+    };
+
+    let dummy_proof_element =
+        Bits256::from_hex_str("0xcafecafecafecafecafecafecafecafecafecafecafecafecafecafecafecafe")
+            .unwrap();
+    let metadata = MultisigMetadata {
+        root: h256_to_bits256(checkpoint.root),
+        index: checkpoint.index,
+        mailbox: h256_to_bits256(checkpoint.mailbox_address),
+        proof: [dummy_proof_element; 32],
+        threshold: TEST_THRESHOLD,
+        signatures: vec![], // Empty, no tests rely upon signatures
```

Comment:
imo we should definitely test the byte offsets with non empty signatures

Reply:
I think bc we're passing in the MultisigMetadata directly rn this doesn't really matter yet, but agreed - just added some signatures